src/clr-angular/utils/drag-and-drop/providers/drag-dispatcher.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Injectable, NgZone, Renderer2} from "@angular/core";
+import {Observable} from "rxjs/Observable";
+import {Subject} from "rxjs/Subject";
+
+import {DragPosition} from "../interfaces/client-position";
+import {Draggable} from "../interfaces/draggable";
+
+@Injectable()
+export class DragDispatcher {
+    private _listeners: Function[];
+
+    draggable: Draggable = {self: null};
+
+    private _dragStartPosition: DragPosition;
+    private _dragEndPosition: DragPosition;
+
+    get dragStartPosition() {
+        return this._dragStartPosition;
+    }
+    get dragEndPosition() {
+        return this._dragEndPosition;
+    }
+
+    private _onDragStart: Subject<MouseEvent|TouchEvent> = new Subject<MouseEvent|TouchEvent>();
+    private _onDragMove: Subject<MouseEvent|TouchEvent> = new Subject<MouseEvent|TouchEvent>();
+    private _onDragEnd: Subject<MouseEvent|TouchEvent> = new Subject<MouseEvent|TouchEvent>();
+
+    get onDragStart(): Observable<MouseEvent|TouchEvent> {
+        return this._onDragStart.asObservable();
+    }
+
+    get onDragMove(): Observable<MouseEvent|TouchEvent> {
+        return this._onDragMove.asObservable();
+    }
+
+    get onDragEnd(): Observable<MouseEvent|TouchEvent> {
+        return this._onDragEnd.asObservable();
+    }
+
+    constructor(private _ngZone: NgZone, private _renderer: Renderer2) {}
+
+    initialize() {
+        if (!this.draggable.handle) {
+            this.draggable.handle = this.draggable.self;
+        }
+
+        this._listeners = [
+            this.customDragEvent(this.draggable.handle, "mousedown", "mousemove", "mouseup"),
+            this.customDragEvent(this.draggable.handle, "touchstart", "touchmove", "touchend")
+        ];
+    }
+
+    customDragEvent(element: Node, startOnEvent: string, moveOnEvent: string, endOnEvent: string): Function {
+        let dragMoveListener: any;
+        let dragEndListener: any;
+
+        return this._renderer.listen(element, startOnEvent, (startEvent: any) => {
+            this._dragStartPosition = {x: startEvent.clientX, y: startEvent.clientY};
+            startEvent.stopPropagation();
```

Comment:
Why was this added? It wasn't there before? I thought we decided that we weren't going to use this in our code.

Reply:
No one else in the application will be able to listen to this event.

Reply:
This was necessary in case users need to nest multiple draggables inside each other, which is common in DnD. Basically, events on children draggables shouldn't interfere with the ones emitted from the parent draggables.

Reply:
I get the reason you did it, but we can't use `stopPropagation()`, simple as that. Use something else to ignore events from nested draggables, but let the event bubble all the way up. A library can never use `stopPropagation()`, it will break any apps or third-party libraries that expect the event to bubble up.